SalesforceCommerceCloud/pwa-kit
# [commerce-sdk-react] Update and invalidate cached queries onSuccess ShopperCustomers Mutations

# Description



The PR implements for each onSuccess ShopperCustomer mutation action the update/invalidate/remove of the affected query keys in the queryClient cache.

All actions in cache are implemented inside the  `onSuccess` callback of the `useShopperCustomersMutation`  hook.

The tests validate:
 1. The mutation action succeeds when the required parameters are provided.
 2. The expected query keys in cache are updated/invalidated/removed.
 

See the list of the affected query keys in the comerce-sdk-react invalidation matrix Quip doc.


# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Implemented the update/invalidate/delete query keys in cache onSuccess ShopperCustomer mutations.
- Tests added to verify the mutation works as expected and the query keys in cache are also updated/invalidated/removed as expected.

# How to Test-Drive This PR

- Run tests:
```
npm run test src/hooks/ShopperCustomers/mutation.test.tsx
```
(Optional) Manually verify the cache changes using React Query Dev Tools
1. Run the test project. ` npm run start --prefix packages/test-commerce-sdk-react/` 
2. Open the useShopperCustomer page. http://localhost:3000/customer
3. Validate the affected query keys in cache, listed in the Quip doc, are invalidated/updated/removed as expected.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
